google-http-client/src/main/java/com/google/api/client/http/UrlEncodedContent.java

```diff
@@ -42,18 +42,33 @@
  *
  * <p>Implementation is not thread-safe.
  *
- * @since 1.0
  * @author Yaniv Inbar
+ * @since 1.0
  */
 public class UrlEncodedContent extends AbstractHttpContent {
 
   /** Key name/value data. */
   private Object data;
 
+  /** Use URI Path encoder flag. False by default (use legacy and deprecated escapeUri) */
+  private Boolean useEscapeUriPathEncoding;
+
   /** @param data key name/value data */
   public UrlEncodedContent(Object data) {
     super(UrlEncodedParser.MEDIA_TYPE);
     setData(data);
+    this.useEscapeUriPathEncoding = false;
+  }
+
+  /**
+   * @param data key name/value data
```

Comment:
the key is a name and a value? That's surprising.

Reply:
Copy paste of the legacy comment (see line 56 of the file.) I kept the legacy signature and behavior, and I added mine with additional parameter. This one is unchanged (and I guess its description also!)

Reply:
Let me know if you want me to improve both or if keeping the existing description is ok.

Reply:
The more I look at this the weirder it is. Why is this an object instead of a Map?

Reply:
I agree; it's strange and I don't know why to use an Object instead of a Map! However to ensure the compatibility with other app that depends on this library, I kept the existing signature.